### ISSUE TITLE
SCF HA mode is not supported for new Secret-Generator function

### ISSUE THREAD

**Issue by zhangtbj** [SUSE/scf#1261]
Hi all,

I am using the latest SCF for my deployment. But I think I found an issue that HA function cannot work fine with new Secret-Generator function.

Because `secret-generation` job will use a property to generate certs in role-manifest:
properties.scf.secrets.variables: '((DOMAIN))((KUBE_SIZING_API_COUNT))((KUBE_SIZING_CC_UPLOADER_COUNT))((KUBE_SIZING_DIEGO_API_COUNT))((KUBE_SIZING_DIEGO_BRAIN_COUNT))((KUBE_SIZING_DIEGO_CELL_COUNT))((KUBE_SIZING_ETCD_COUNT))'

But for example `KUBE_SIZING_DIEGO_CELL_COUNT` is got from count setting in values.yml.

But if I set the `HA: true` in values.yml but not change the count setting of diego-cell, SCF will still use the count setting instead.

For example, the initial size of diego-cell is 1, I use HA mode and set HA count is 6, but secret-generation will still use 1 as the count of diego-cell. 

Then only diego-cell-0 works fine and others will fail at auction:

```
{"timestamp":"1514257592.442109346","source":"auctioneer","message":"auctioneer.auction.failed-to-get-state","log_level":2,"data":{"cell-guid":"diego-cell-4","error":"Get https://diego-cell-4.diego-cell-set.cf.svc.cluster.local:1801/state: x509: certificate is valid for diego-cell, *.diego-cell, diego-cell.cf.svc, *.diego-cell.cf.svc, diego-cell.cf.svc.cluster.local, *.diego-cell.cf.svc.cluster.local, diego-cell-0.diego-cell-set, diego-cell-0.diego-cell-set.cf.svc, diego-cell-0.diego-cell-set.cf.svc.cluster.local, diego-cell.cf.svc.cluster.local, *.diego-cell.cf.svc.cluster.local, rep_server, not diego-cell-4.diego-cell-set.cf.svc.cluster.local","session":"629146"}}
```

I am not sure if anyone uses HA mode, only HA and non-HA mode is not enough for the end user multiple choice I think.

Pls let me know if I understand wrong.

Thanks! :)

**Comment by aarondl** [comment]
We're looking at this soon! Thanks for bringing it up.

**Comment by zhangtbj** [comment]
Hi @aarondl Thanks!

And we are using another function for the different size of deployment:
```
helm install output/helm \
     -f ./config/cf_enlarge_counts.yaml \
```

We generate a patch file `cf_enlarge_counts.yaml` to set size for each component and use `-f` in helm install, then we can set different size. I think it is better and more flexible than HA mode.

We have verified it works fine with Secret-Generation function.

Hope it is helpful. :)

**Comment by mook-as** [comment]
This should now be fixed via #1266
Though we should probably still reconsider the `HA: true` flag… using an extra file sounds better, but that wouldn't work if you `helm install` using a chart reference after adding a helm repo that contains the chart. That's (I'm pretty sure) why we had that odd HA mode in the first place…

Thanks again for reporting this, @zhangtbj :)
